example-app/src/main/java/com/hcaptcha/example/MainActivity.java

```diff
@@ -14,22 +21,65 @@
 
     private static final String TAG = MainActivity.class.getSimpleName();
 
+    private RadioGroup sizeRadioGroup;
+    private CheckBox setupCheckBox;
     private TextView tokenTextView;
-
     private TextView errorTextView;
 
+    private HCaptcha hCaptcha;
+    private HCaptchaConfig hCaptchaConfig;
+
+    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        this.tokenTextView = findViewById(R.id.tokenTextView);
-        this.errorTextView = findViewById(R.id.errorTextView);
-        // For debugging purposes only
-        // android.webkit.WebView.setWebContentsDebuggingEnabled(true);
+        setupCheckBox = findViewById(R.id.setupCheckBox);
+        setupCheckBox.setOnCheckedChangeListener((checkBox, checked) -> {
+            initHCaptcha(getSizeFromRadio(sizeRadioGroup));
+        });
+
+        sizeRadioGroup = findViewById(R.id.sizeRadioGroup);
+        sizeRadioGroup.setOnCheckedChangeListener((radioGroup, resId) -> {
+            initHCaptcha(getSizeFromRadio(radioGroup));
+        });
+        sizeRadioGroup.check(R.id.size_invisible);
+        tokenTextView = findViewById(R.id.tokenTextView);
+        errorTextView = findViewById(R.id.errorTextView);
+
+        CheckBox debugCheckBox = findViewById(R.id.webViewDebug);
+        debugCheckBox.setOnCheckedChangeListener((checkBox, checked) -> {
+            android.webkit.WebView.setWebContentsDebuggingEnabled(checked);
+        });
+    }
+
+    private HCaptchaSize getSizeFromRadio(@NonNull RadioGroup radioGroup) {
+        switch (radioGroup.getCheckedRadioButtonId()) {
+            case R.id.size_invisible: return HCaptchaSize.INVISIBLE;
+            case R.id.size_compact: return HCaptchaSize.COMPACT;
+            case R.id.size_normal: return HCaptchaSize.NORMAL;
+            default: return null;
+        }
+    }
+
+    private void initHCaptcha(@Nullable HCaptchaSize size) {
+        HCaptchaConfig.HCaptchaConfigBuilder builder = HCaptchaConfig.builder()
+                .siteKey("10000000-ffff-ffff-ffff-000000000001")
+                .loading(true);
+        if (size != null) {
+            builder.size(size);
+        }
+
+        hCaptchaConfig = builder.build();
+
+        hCaptcha = HCaptcha.getClient(this);
+        if (setupCheckBox.isChecked()) {
+            hCaptcha.setup(hCaptchaConfig);
+        }
     }
 
     private void setTokenTextView(final String text) {
-        tokenTextView.setText(text.substring(0, Math.min(100, text.length())) + "...");
```

Comment:
This change shows the whole token, but there is no scrolling, thus it looks odd. Why the change?

![image](https://user-images.githubusercontent.com/11096226/162221777-fc8b5a49-e5e7-4304-b4c4-d2741dec8c74.png)

Reply:
Fixed